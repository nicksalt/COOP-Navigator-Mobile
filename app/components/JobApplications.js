import React, { Component } from "react";
import { FlatList, StyleSheet, Text, View } from "react-native";
import JobListItem from "./JobListItem";

export default class JobApplications extends Component {
  componentDidMount = () => {
    this.props.fetchJobApplicationsDispatch();
  };

  render() {
    if (this.props.errorOccurred) {
      return (
        <View>
          <Text>There was an error fetching your applications:</Text>
          <Text>{this.props.error.message}</Text>
        </View>
      );
    }
    return (
      <FlatList
        data={this.props.jobs}
        keyExtractor={job => job.jobId.toString()}
        refreshing={!!this.props.isFetching}
        onRefresh={this.props.fetchJobApplicationsDispatch}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
        ListEmptyComponent={
          !this.props.isFetching && (
            <Text style={styles.centerMsg}>
              You have not applied to any jobs.
            </Text>
          )
        }
        renderItem={({ item }) => (
          <JobListItem
            job={item}
            navigation={this.props.navigation}
            fetchSingleJobDispatch={this.props.fetchSingleJobDispatch}
          />
        )}
      />
    );
  }
}

const styles = StyleSheet.create({
  separator: {
    height: 1,
    backgroundColor: "#ccc"
  },
  centerMsg: {
    padding: 5,
    fontSize: 13,
    textAlign: "center",
    color: "black",
    margin: 2
  }
});
