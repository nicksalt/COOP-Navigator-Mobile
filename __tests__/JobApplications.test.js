import React from "react";
import JobApplications from "../app/components/JobApplications";
import renderer from "react-test-renderer";

jest.mock("FlatList", () => "FlatList");

test("JobApplications snapshot renders correctly", () => {
  let props = {};
  props.fetchJobApplicationsDispatch = () => {};
  props.isFetching = true;
  let tree = renderer.create(<JobApplications {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.isFetching = false;
  props.errorOccurred = true;
  props.error = { message: "ERROR OCCURRED" };
  tree = renderer.create(<JobApplications {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.errorOccurred = false;
  props.jobs = [];
  tree = renderer.create(<JobApplications {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
