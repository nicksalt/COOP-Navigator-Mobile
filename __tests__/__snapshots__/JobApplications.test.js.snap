// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobApplications snapshot renders correctly 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={false}
  keyExtractor={[Function]}
  onRefresh={[Function]}
  refreshing={true}
  renderItem={[Function]}
/>
`;

exports[`JobApplications snapshot renders correctly 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    There was an error fetching your applications:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    ERROR OCCURRED
  </Text>
</View>
`;

exports[`JobApplications snapshot renders correctly 3`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "margin": 2,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      You have not applied to any jobs.
    </Text>
  }
  data={Array []}
  keyExtractor={[Function]}
  onRefresh={[Function]}
  refreshing={false}
  renderItem={[Function]}
/>
`;
